src/pal/src/loader/module.cpp

```diff
@@ -1339,29 +1342,59 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
     // See if file can be dlopen()ed; this should work even if it's already loaded
     {
         // See GetProcAddress for an explanation why we leave the PAL.
-        PAL_LeaveHolder holder;
-        dl_handle = dlopen(shortAsciiName, RTLD_LAZY | RTLD_NOLOAD); 
-        if (!dl_handle)
+        PAL_LeaveHolder holder; 
+        
+        // Try to dlopen the module with appropriate variations on the name.
+        // P/Invokes are often declared with shortened versions of the real library
+        // name, such as without the extension or without the "lib" prefix.
+        // We try to open with such variations on the original.
+        int endsWithSuffix = FALSE, startsWithPrefixOrIsAbsolute = FALSE;
+        size_t shortAsciiNameLen = 0;
+        int i;
+        for (i = 1; i <= 4; i++) // four cases: original, with prefix, with suffix, with both prefix and suffix
         {
-            dl_handle = dlopen(shortAsciiName, RTLD_LAZY);
-        }
+            const char* filename = NULL;
+            switch (i)
+            {
+                case 1: // original name
+                    filename = shortAsciiName;
+                    break;
+
+                case 2: // with prefix prepended
```

Comment:
My opinion on the matter is to take the string and add prefix and suffix regardless of whether they exist or not.  Any 'name mangling' logic other than that should exist in the future solution  for #930.  This simplifies the runtime, eliminates false positives, and is deterministic independent of what file name may exist or not in the library working directory or search path.  Additionaly, it makes mapping names in the future much simpler as multiple permutations do not need to be accounted for.

While there are (many) exceptions to the rule, prefix+library_name+suffix is canonical.  The exceptions and eccentricities should be handled by the solution to #930.  This should handle the majority of cases without any further work.